### PR TITLE
blocks: Use templates more efficiently in add_blk

### PR DESCRIPTION
**Pull request by jdemel**
Compare recent changes in the moving_average block with those templatized
`volk_add` functions instead of whole work functions being templatized.

It should now be easier to add further templates for other block types.
A volkized version for gr_complex was added.

### CODE REVIEW

**Review by marcusmueller — APPROVED** [review]

**Comment by marcusmueller** [comment]
Nice; invalidates my #3899, but that's fine with me. The chanest test case fails for me, too. It's wonky, for some reason.

**Comment by jdemel** [comment]
CI says this PR is fine. What should we do with the test that fails for you? Ignore it as an outlier?

File: gr-blocks/lib/add_blk_impl.cc [hunk 1]
```diff
@@ -34,25 +57,19 @@ add_blk_impl<float>::add_blk_impl(size_t vlen)
...
-    const int alignment_multiple = volk_get_alignment() / sizeof(float);
-    set_alignment(std::max(1, alignment_multiple));
+    set_alignment(std::max(1, int(volk_get_alignment() / sizeof(float))));
 }
 
 template <>
-int add_blk_impl<float>::work(int noutput_items,
-                              gr_vector_const_void_star& input_items,
-                              gr_vector_void_star& output_items)
+add_blk_impl<gr_complex>::add_blk_impl(size_t vlen)
```
Comment (rear1019) [inline gr-blocks/lib/add_blk_impl.cc:64]: Just wondering: Is this specialization of the constructor (and the one for `float`) required? Was it ever required? (As per C++ rules – my knowledge of templates is limited…)

**Review by rear1019 — APPROVED** [review]

**Comment by mbr0wn** [comment]
Three approvals and I can't repro the CI issue... so it's going in.

**Pull request merged**